`jx step changelog` is breaking all master builds of JX

```
jx step changelog  --header-file docs/dev/changelog-header.md --version 1.3.753
Using batch mode as inside a pipeline
Generating change log from git ref v1.3.751 => 5a973888457b62d240dee12b645eb527ded0c3d4
Failed to enrich commits with issues: id cannot be empty
Failed to enrich commits with issues: id cannot be empty
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x204812c]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/jx/cmd.(*StepChangelogOptions).addCommit(0xc0009a0e00, 0xc000314a68, 0xc002376f00, 0xc000a61620)
    /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/step_changelog.go:531 +0x20c
github.com/jenkins-x/jx/pkg/jx/cmd.(*StepChangelogOptions).Run(0xc0009a0e00, 0x0, 0x0)
    /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/step_changelog.go:360 +0x3327
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdStepChangelog.func1(0xc000926f00, 0xc000997f00, 0x0, 0x4)
    /home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/step_changelog.go:161 +0x70
github.com/spf13/cobra.(*Command).execute(0xc000926f00, 0xc000997ec0, 0x4, 0x4, 0xc000926f00, 0xc000997ec0)
    /home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004f8a00, 0x0, 0x0, 0x0)
    /home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0xc0004f8a00, 0xc0004dd770, 0x30ef8e0)
    /home/jenkins/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc000a0bf88, 0xc0000c4058)
    /home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/app/jx.go:17 +0xed
main.main()
    /home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/jx.go:11 +0x22
Makefile:163: recipe for target 'release' failed
make: *** [release] Error 2
[Pipeline] }
```

--- a/pkg/jx/cmd/step_changelog.go
+++ b/pkg/jx/cmd/step_changelog.go
@@ -524,13 +524,20 @@ func (o *StepChangelogOptions) addCommit(spec *v1.ReleaseSpec, commit *object.Co
 	if err != nil {
 		log.Warnf("Failed to enrich commits with issues: %v\n", err)
 	}
+	var authorDetails, committerDetails v1.UserDetails
+	if author != nil {
+		authorDetails = author.Spec
+	}
+	if committer != nil {
+		committerDetails = committer.Spec
+	}
 	commitSummary := v1.CommitSummary{
 		Message:   commit.Message,
 		URL:       url,
 		SHA:       sha,
-		Author:    &author.Spec,
+		Author:    &authorDetails,
 		Branch:    branch,
-		Committer: &committer.Spec,
+		Committer: &committerDetails,
 	}
 	err = o.addIssuesAndPullRequests(spec, &commitSummary, commit)
 

--- a/pkg/users/git.go
+++ b/pkg/users/git.go
@@ -49,8 +49,14 @@ func (r *GitUserResolver) GitUserSliceAsUserDetailsSlice(users []gits.GitUser) (
 func (r *GitUserResolver) Resolve(user *gits.GitUser) (*jenkinsv1.User, error) {
 	selectUsers := func(id string, users []jenkinsv1.User) (string, []jenkinsv1.User,
 		*jenkinsv1.User, error) {
+		var gitUser *gits.GitUser
+		if user.Login != "" {
+			gitUser = r.GitProvider.UserInfo(user.Login)
+		}
+		if gitUser == nil {
+			gitUser = user
+		}
 
-		gitUser := r.GitProvider.UserInfo(user.Login)
 		possibles := make([]jenkinsv1.User, 0)
 		if gitUser == nil {
 			// annoyingly UserInfo swallows the error, so we recreate it!
@@ -62,10 +68,8 @@ func (r *GitUserResolver) Resolve(user *gits.GitUser) (*jenkinsv1.User, error) {
 				}
 			}
 		}
-		new := r.GitUserToUser(mergeGitUsers(gitUser, user))
-		if gitUser != nil {
-			id = gitUser.Login
-		}
+		new := r.GitUserToUser(gitUser)
+		id = gitUser.Login
 		return id, possibles, new, nil
 	}
 	return Resolve(user.Login, r.GitProviderKey(), r.JXClient, r.Namespace, selectUsers)

--- a/pkg/users/git_test.go
+++ b/pkg/users/git_test.go
@@ -1,1 +1,199 @@
-package users
+package users_test
+
+import (
+	"fmt"
+	"testing"
+
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+
+	"github.com/jenkins-x/jx/pkg/users"
+
+	"github.com/satori/go.uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jenkinsv1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/helm/mocks"
+	"github.com/jenkins-x/jx/pkg/jx/cmd"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestFindUserByLabel(t *testing.T) {
+	resolver, _, err := prepare(t)
+	assert.NoError(t, err)
+	gitUserID := uuid.NewV4().String()
+	// Create the user
+	labeledUserID, err := createDummyUser(resolver, true, gitUserID)
+	defer func() {
+		err := removeDummyUser(labeledUserID, resolver)
+		assert.NoError(t, err)
+	}()
+	assert.NoError(t, err)
+
+	unLabeledUserID, err := createDummyUser(resolver, false, gitUserID)
+	defer func() {
+		err := removeDummyUser(unLabeledUserID, resolver)
+		assert.NoError(t, err)
+	}()
+	assert.NoError(t, err)
+	gitUser := gits.GitUser{
+		Login: gitUserID,
+	}
+	user, err := resolver.Resolve(&gitUser)
+	assert.NoError(t, err)
+	// Validate that we found the labeled one, not the unlabeled one
+	assert.Equal(t, labeledUserID, user.Spec.Login)
+}
+
+func TestFindUserBySignature(t *testing.T) {
+	resolver, _, err := prepare(t)
+	assert.NoError(t, err)
+	gitUserID := uuid.NewV4().String()
+	// Create the user
+	userID, err := createDummyUser(resolver, true, gitUserID)
+	defer func() {
+		err := removeDummyUser(userID, resolver)
+		assert.NoError(t, err)
+	}()
+	assert.NoError(t, err)
+	assert.NoError(t, err)
+	signature := object.Signature{
+		Email: fmt.Sprintf("%s@test.com", userID),
+		Name:  userID,
+	}
+	user, err := resolver.GitSignatureAsUser(&signature)
+	assert.NoError(t, err)
+	// Validate that we found the labeled one, not the unlabeled one
+	assert.Equal(t, userID, user.Spec.Login)
+}
+
+func TestFindUserByAccountReference(t *testing.T) {
+	resolver, _, err := prepare(t)
+	assert.NoError(t, err)
+	gitUserID1 := uuid.NewV4().String()
+	gitUserID2 := uuid.NewV4().String()
+	// Create the user
+	userID1, err := createDummyUser(resolver, false, gitUserID1)
+	defer func() {
+		err := removeDummyUser(userID1, resolver)
+		assert.NoError(t, err)
+	}()
+	assert.NoError(t, err)
+
+	userID2, err := createDummyUser(resolver, false, gitUserID2)
+	defer func() {
+		err := removeDummyUser(userID2, resolver)
+		assert.NoError(t, err)
+	}()
+	assert.NoError(t, err)
+	gitUser := gits.GitUser{
+		Login: gitUserID1,
+	}
+	user, err := resolver.Resolve(&gitUser)
+	assert.NoError(t, err)
+	// Validate that we found the correct user
+	assert.Equal(t, userID1, user.Spec.Login)
+	// Validate that the label was added
+	val, ok := user.Labels[resolver.GitProviderKey()]
+	assert.True(t, ok)
+	assert.Equal(t, gitUserID1, val)
+}
+
+func TestFindUserByFromGitProvider(t *testing.T) {
+	resolver, fakeProvider, err := prepare(t)
+	assert.NoError(t, err)
+	gitUserID1 := uuid.NewV4().String()
+	gitUserID2 := uuid.NewV4().String()
+	// Create the user
+	gitUser1 := &gits.GitUser{
+		Name:  uuid.NewV4().String(),
+		Email: fmt.Sprintf("%s@test.com", uuid.NewV4().String()),
+		Login: gitUserID1,
+	}
+	fakeProvider.Users = []*gits.GitUser{
+		gitUser1,
+	}
+	userID2, err := createDummyUser(resolver, false, gitUserID2)
+	defer func() {
+		err := removeDummyUser(userID2, resolver)
+		assert.NoError(t, err)
+	}()
+	assert.NoError(t, err)
+	gitUser := gits.GitUser{
+		Login: gitUserID1,
+	}
+	user, err := resolver.Resolve(&gitUser)
+	assert.NoError(t, err)
+	// Validate that we create a new user from the git provider
+	assert.Equal(t, gitUser1.Email, user.Spec.Email)
+	assert.Equal(t, gitUser1.Name, user.Spec.Name)
+	// Validate that the label was added
+	val, ok := user.Labels[resolver.GitProviderKey()]
+	assert.True(t, ok)
+	assert.Equal(t, gitUserID1, val)
+	// Validate that the account reference is created
+	assert.Len(t, user.Spec.Accounts, 1)
+	assert.Equal(t, user.Spec.Accounts[0].Provider, resolver.GitProviderKey())
+	assert.Equal(t, user.Spec.Accounts[0].ID, gitUserID1)
+}
+
+func prepare(t *testing.T) (*users.GitUserResolver, *gits.FakeProvider, error) {
+	testOrgName := "myorg"
+	testRepoName := "my-app"
+	fakeRepo := gits.NewFakeRepository(testOrgName, testRepoName)
+
+	o := cmd.CommonOptions{}
+	cmd.ConfigureTestOptionsWithResources(&o,
+		[]runtime.Object{},
+		[]runtime.Object{},
+		&gits.GitFake{},
+		helm_test.NewMockHelmer(),
+	)
+
+	jxClient, ns, err := o.CreateJXClient()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	fakeProvider := gits.NewFakeProvider(fakeRepo)
+	fakeProvider.Type = gits.Fake
+	return &users.GitUserResolver{
+		GitProvider: fakeProvider,
+		JXClient:    jxClient,
+		Namespace:   ns,
+	}, fakeProvider, nil
+}
+
+func createDummyUser(resolver *users.GitUserResolver, createLabels bool, gitUserID string) (string, error) {
+	id := uuid.NewV4().String()
+	spec := jenkinsv1.UserDetails{
+		Name:  id,
+		Email: fmt.Sprintf("%s@test.com", id),
+		Login: id,
+		Accounts: []jenkinsv1.AccountReference{
+			jenkinsv1.AccountReference{
+				ID:       gitUserID,
+				Provider: resolver.GitProviderKey(),
+			},
+		},
+	}
+	meta := metav1.ObjectMeta{
+		Name: id,
+	}
+	if createLabels {
+		meta.Labels = map[string]string{
+			resolver.GitProviderKey(): gitUserID,
+		}
+	}
+	_, err := resolver.JXClient.JenkinsV1().Users(resolver.Namespace).Create(&jenkinsv1.User{
+		Spec:       spec,
+		ObjectMeta: meta,
+	})
+	return id, err
+}
+
+func removeDummyUser(id string, resolver *users.GitUserResolver) error {
+	return resolver.JXClient.JenkinsV1().Users(resolver.Namespace).Delete(id, &metav1.DeleteOptions{})
+}

--- a/pkg/users/users.go
+++ b/pkg/users/users.go
@@ -87,9 +87,6 @@ func DeleteUser(jxClient versioned.Interface, ns string, userName string) error 
 func Resolve(id string, providerKey string, jxClient versioned.Interface,
 	namespace string, selectUsers func(id string, users []jenkinsv1.User) (string,
 		[]jenkinsv1.User, *jenkinsv1.User, error)) (*jenkinsv1.User, error) {
-	if id == "" {
-		return nil, fmt.Errorf("id cannot be empty")
-	}
 	if id != "" {
 
 		labelSelector := fmt.Sprintf("%s=%s", providerKey, id)

